### PR TITLE
Use range based looping

### PR DESCRIPTION
**Pull request by Nirma**
Thought this way of writing the loops was a bit more modern. 

Hope it helps!

### CODE REVIEW

File: ReactiveCocoa/Swift/Bag.swift [hunk 1]
```diff
@@ -60,7 +60,7 @@ internal struct Bag<T> {
 	/// will reset all current identifiers to reclaim a contiguous set of
 	/// available identifiers for the future.
 	private mutating func reindex() {
-		for var i = 0; i < elements.count; i++ {
+		for i in 0..<elements.count {
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Bag.swift]: This is already better, but there's an even safer way of creating this `Range`: using [`indices`](http://swiftdoc.org/func/indices/).

Comment (Nirma) [inline ReactiveCocoa/Swift/Bag.swift]: @NachoSoto Thanks for the tip! I swapped out count with endIndex. 
For the startIndex I think leaving that as 0 is fine but endIndex is much better than what I had before with using count.

Comment (NachoSoto) [inline ReactiveCocoa/Swift/Bag.swift]: It's only fine because it's an array, though `indices` would be more correct.
But sure, this is okay.

**Comment by NachoSoto** [comment]
Failing test is unrelated

**Pull request merged**